### DISCUSSION TITLE
Is Coil's image cache encrypted and decrypted automatically?

### DISCUSSION THREAD

**Discussion by rickycaron** [coil-kt/coil#1801] [General]
I am going to use Coil recently in a project. The requirement is all the files need to be encrypted. Does Coil provide such encryption? Or does it provide an API for developers to customize encryption and decryption?

**Comment by colinrtwhite** [comment]
Coil's cache isn't encrypted. You could intercept calls to `DiskCache`'s methods (as it's an interface) to encrypt/decrypt image data when it's being read/written from the disk cache, though this would have to be custom and isn't supported out of the box.
